server/artifacts/artifact_server.go

```diff
@@ -84,6 +86,174 @@ func (a *ArtifactServer) GetInputArtifact(w http.ResponseWriter, r *http.Request
 	a.getArtifact(w, r, true)
 }
 
+// single endpoint to be able to handle serving directories as well as files, both those that have been archived and those that haven't
+// Valid requests:
+//  /artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/outputs/{artifactName}
+//  /artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/outputs/{artifactName}/{fileName}
+//  /artifact-files/{namespace}/[archived-workflows|workflows]/{id}/{nodeId}/outputs/{artifactName}/{fileDir}/.../{fileName}
+// 'id' field represents 'uid' for archived workflows and 'name' for non-archived
+func (a *ArtifactServer) GetArtifactFile(w http.ResponseWriter, r *http.Request) {
+
+	const (
+		NAMESPACE_INDEX       = 2
```

Comment:
Minor - Golang convention for const is different to Java. They should be camel-case.`namespaceIndex`